test/end-to-end/company.js

```diff
@@ -100,6 +100,32 @@ describe("Company application endpoint", () => {
                 fs.unlinkSync(filename);
             });
 
+            test("should finish the application with single contact", async () => {
+                await test_agent
+                    .post("/company/application/finish")
+                    .attach("logo", "test/data/logo-niaefeup.png")
+                    .field("bio", "A very interesting and compelling bio")
+                    .field("contacts", "contact1")
+                    .expect(HTTPStatus.OK);
+
+                const test_company = [... await Company.find({}).exec()][0];
```

Comment:
define test_company on the describe level (outside of the beforeEach) and here you won't need to query the db, making the test faster, as you can access test_company directly.

You should do this (also remove the .exec) on the test above

Reply:
The company is updated in each test (contacts are added, etc), therefore a new db query is necessary